### ISSUE TITLE
Analyzing modules with information from local library

### ISSUE THREAD

**Issue by erivas** [rescript-association/reanalyze#120]
I have a project which has the "usual" `bin`/`lib` structure. The binary uses the library in `lib`, which exposes a function that raises an exception. Is there a way to make the exception analyzer be aware that the code calling this function raises an exception?

I have written the example [here](https://github.com/erivas/rexample/tree/c77cd350030e989e46e050393c81b9c1e8ca9d45). More concretely, I think I would like `  hello_world.ml:8:10 Call(Mylib.Lib.trace, modulePath:) ` to use the information from the corresponding `lib.ml` that was already processed, and warn that it might raise `Bad`. Is there a way of doing this? Thanks!

**Comment by cristianoc** [comment]
@erivas This ended up in master instead of a PR by mistake: https://github.com/rescript-association/reanalyze/commit/cdfda77abac5defd1271927bcd277628afa7e36a

It's pretty basic: when looking for `MyLib.Foo` it also tries `Foo`. The analysis does not try to do advanced path resolution. But it should work for the cases as in your example.

**Comment by erivas** [comment]
Wow! Thanks for the quick fix! it indeed solves the example I proposed.

I've updated the [example](https://github.com/erivas/rexample) with a setting closer to my real application. I am having the following problem: two files are named the same (in different libraries, in the example `lib` and `otherlib`), it seems that the analysis of one of them gets mixed with the analysis of the other: it kind of breaks the analysis of `lib`,
```shell
  Events combine: #events 1
  lib.ml:3:28 raises  Bad

  
  Events combine: #events 1
  lib.ml:5:13 Call(trace, modulePath:) 
```
but doesn't report that `lib.ml:5` will raise an exception. Does the problem appear in the `Scanning` phase when building the table?
